test-data/unit/check-based-default-return.test

```diff
@@ -0,0 +1,103 @@
+-- Type checker test cases for default return type.
+
+
+[case testSimple]
+# flags: --default-return
+def f(): ...
+reveal_type(f)  # N: Revealed type is "def ()"
+
+def g(i: int): ...
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int)"
+
+class A:
+    def f(self): ...
+    def g(self, i: int): ...
+
+reveal_type(A.f)  # N: Revealed type is "def (self: __main__.A)"
+reveal_type(A.g)  # N: Revealed type is "def (self: __main__.A, i: builtins.int)"
+
+
+[case testUntypedDefs]
+# flags: --default-return --allow-untyped-defs --allow-any-expr
+
+def f(): ...
+reveal_type(f)  # N: Revealed type is "def () -> Any"
+
+def g(i: int): ...
+reveal_type(g)  # N: Revealed type is "def (i: builtins.int)"
+
+def h(i: int, j): ... # E:
+reveal_type(h)  # N: Revealed type is "def (i: builtins.int, j: Any) -> Any"
+
+class A:
+    def f(self): ...
+    def g(self, i: int): ...
+    def h(self, i: int, j): ... # E:
```

Comment:
What’s the diff between untyped defs and incomplete defs. Id assume this is an incomplete def?

Reply:
it is, and it's expecting an error